src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowLoopSpec.groovy

```diff
@@ -0,0 +1,772 @@
+        withPool {
+            [exam.forward, exam.reverse].eachParallel { direction ->
+                def resources = traffExam.startExam(direction)
+                direction.setResources(resources)
+                assert !traffExam.waitExam(direction).hasTraffic()
+            }
+        }
+
+        and: "Counter on the flowLoop rules are increased"
+        getFlowLoopRules(switchPair.src.dpId)*.packetCount.every { it > 0 }
+
+        cleanup: "Revert system to original state"
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        if (!portIsDown) {
+            antiflap.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+            Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+                assert islUtils.getIslInfo(islToBreak).get().state == IslChangeType.DISCOVERED
+            }
+        }
+        database.resetCosts()
+    }
+
+    def "Unable to create flowLoop for a singleSwitch flow"() {
+        given: "An active singleSwitch flow"
+        def sw = topology.activeSwitches.first()
+        def flow = flowHelperV2.singleSwitchFlow(sw)
+        flowHelperV2.addFlow(flow)
+
+        when: "Create flowLoop on the sw switch"
+        northboundV2.createFlowLoop(flow.flowId, new FlowLoopPayload(sw.dpId))
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpServerErrorException)
+        exc.statusCode == HttpStatus.NOT_IMPLEMENTED
+        def errorDetails = exc.responseBodyAsString.to(MessageError)
+        errorDetails.errorMessage == "Could not update flow"
+        errorDetails.errorDescription == "Loop for single switch flows is not implemented"
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+    }
+
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    def "Unable to create flowLoop for a QinQ flow"() {
+        given: "An active QinQ flow"
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].every { northbound.getSwitchProperties(it.dpId).multiTable }
+        } ?: assumeTrue("Not able to find enough switches in multi-table mode", false)
+        def flow = flowHelperV2.randomFlow(swP)
+        flow.source.innerVlanId = 4093
+        flow.destination.innerVlanId = 3904
+        northboundV2.addFlow(flow)
+
+        when: "Create flowLoop on the src switch"
+        northboundV2.createFlowLoop(flow.flowId, new FlowLoopPayload(swP.src.dpId))
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpServerErrorException) //bug TODO: create an issue
+        exc.statusCode == HttpStatus.NOT_IMPLEMENTED
+        def errorDetails = exc.responseBodyAsString.to(MessageError)
+        errorDetails.errorMessage == "Could not update flow"
+        errorDetails.errorDescription == "Loop for Q-in-Q flows is not implemented"
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+    }
+
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    def "Unable to create flowLoop twice on the same flow"() {
```

Comment:
`"Unable to create flowLoop both on src and dst at the same time"` describes it better

Reply:
I also think that below test `def "Unable to create flowLoop on the src switch when it is already created on the dst switch"() {` duplicates this one